lib/pbench/server/api/resources/server_configuration.py

```diff
@@ -0,0 +1,152 @@
+from http import HTTPStatus
+from logging import Logger
+
+from flask.json import jsonify
+from flask.wrappers import Request, Response
+
+from pbench.server import PbenchServerConfig
+from pbench.server.api.resources import (
+    API_AUTHORIZATION,
+    API_METHOD,
+    APIAbort,
+    API_OPERATION,
+    ApiBase,
+    ApiParams,
+    ApiSchema,
+    MissingParameters,
+    ParamType,
+    Parameter,
+    Schema,
+)
+from pbench.server.database.models.server_config import (
+    ServerConfig,
+    ServerConfigBadValue,
+    ServerConfigError,
+)
+
+
+class ServerConfiguration(ApiBase):
+    """
+    API class to retrieve and mutate Dataset metadata.
+    """
+
+    def __init__(self, config: PbenchServerConfig, logger: Logger):
+        super().__init__(
+            config,
+            logger,
+            ApiSchema(
+                API_METHOD.PUT,
+                API_OPERATION.UPDATE,
+                uri_schema=Schema(
+                    Parameter("key", ParamType.KEYWORD, keywords=ServerConfig.KEYS)
+                ),
+                # TODO: We aspire to be flexible about how a config option is
+                # specified on a PUT. While the "config" parameter is specifically
+                # intended to support a full JSON body that sets or updates all
+                # options at once, for individual options using the "value"
+                # query parameter is probably easier... but the "value" body
+                # parameter also allows specifying the value in a body.
+                #
+                #   PUT /server/configuration/dataset-lifetime?value=2y
+                #   PUT /server/configuration/dataset-lifetime
+                #       {"value": "2y"}
+                #   PUT /server/configuration/dataset-lifetime ???
+                #       {"config": {"dataset-lifetime": "2y"}}
```

Comment:
This case seems like it would be better handled by 
```
  PUT /server/configuration/
    {"dataset-lifetime": "2y"}
```
That is, if the `configuration` API is hit with a null key value, then look for the key-value pair in the payload.  (Does Flask support this?...I can't remember....)

This would enable reloading the whole configuration in a single request (which might be handy or even important):
```
  PUT /server/configuration/
        {
           "dataset-lifetime": "3650",
           "server-state": {"status": "enabled"},
           "server-banner": None,
        }
```

Reply:
When I was playing with possible syntax, I wanted to cover three main cases. The third entry here, which was marked with the "???", was a degenerate case of the third case I really intended, and coded. You can set a single parameter either using `/server/configuration/name?value=string` or `/server/configuration/name` with `{"value": "string"}`. The "config" JSON key was really intended for setting multiple options, as you suggest in your example ... and that's how it works, and I guess my documentation-brain got side-tracked by this weird case so I didn't finish.

The question here was whether I should allow the kinda odd combination of `/server/configuration/name` with `{"config": {"name": "string"}}`. I didn't code that, though I thought about it. Really it makes more sense when you drop the `name` from the URL... which is why I didn't code this case.

I should have fixed the docstring, and had I started that I probably would have noticed I'd failed to document the *real* third case. 🙄